### ISSUE TITLE
GSPlugin doesn't work

### ISSUE THREAD

**Issue by CXianRen** [Cecile-hi/Multimodal-Learning-with-Alternating-Unimodal-Adaptation#14]
Hi, thanks for sharing the code. 

Well, with my test, GSPlugin doesn't work while training. 

MLA passes the fc layer to the gsplugin: 

https://github.com/Cecile-hi/Multimodal-Learning-with-Alternating-Unimodal-Adaptation/blob/ab6c184579f9ff390e9c9b8216c9900c2a43545b/main.py#L437

But the code:
```py
    def before_update(self, model, before_batch_input, batch_index, len_dataloader, train_exp_counter):

        lamda = batch_index / len_dataloader + 1
        alpha = 1.0 * 0.1 ** lamda
        # x_mean = torch.mean(strategy.mb_x, 0, True)
        if train_exp_counter != 0:
            for n, w in model.named_parameters():

                if n == "module.weight":

```

I think it should be 
```py
if n == "weight":
```
With this fix, gsplugin work. ! But this method doesn't work, at least on myside. By the way, it also causes memory leak.

Using parameters:
--batch_size 16
CREMAD
others are default.

Without fix:
![image](https://github.com/user-attachments/assets/ed3d5736-25bc-4185-9a18-6372c6a1f822)

With fix:

![image](https://github.com/user-attachments/assets/021c7130-0af7-47be-be9a-04865221a790)

**Comment by CXianRen** [comment]
Actually:
1.1 the code computing entropy and weight doesn't work for when batch size > 1 (only for batch size ==1 is ok)
2. Use logit output instead of Softmax  output for fusing get result close to the paper result. (so I doubt you have a mistake in your paper)
3. I doubt your GS plugin is meaningless, because share head is only 1 linear layer. because using without it get much higher acc.
